### PR TITLE
[Test] Traffic is not recorded by default in resource preparers

### PR DESCRIPTION
**Pull request by adewaleo**
Add new GraphClient replacer. Update preparers. GeneralNameReplacer. Re-record tests.
Properly handle python3 bytes.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by adewaleo** [comment]
Not requesting review yet as this is still WIP

**Comment by adewaleo** [comment]
I was able to find the cause for #9591 when I attempted these changes previously. I would feel more comfortable running these checks if I new that all the scenario tests are being run in the CI

**Comment by tjprescott** [comment]
Given how late in the sprint it is, I recommend deferring this merge until after the release branch is cut. cc/ @yugangw-msft

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/recordings/test_functionapp_slot_appsetting_update.yaml [hunk 1]
```diff
@@ -1,32 +1,27 @@
...
-      "date": "2019-06-19T22:02:05Z"}}'
+    body: null
     headers:
       Accept:
       - application/json
       Accept-Encoding:
       - gzip, deflate
       CommandName:
-      - group create
+      - functionapp plan create
```
Comment (adewaleo) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/recordings/test_functionapp_slot_appsetting_update.yaml]: As we can see here and [here](https://github.com/Azure/azure-cli/pull/9747/files#diff-8ed9aaa34535ded5e7137146ba395147R56) the first few requests is not `group create` or `storage account create` but `functionapp plan create`

**Comment by yugangw-msft** [comment]
@adewaleo, can this be merged now?

**Pull request merged**